src/Http/Http.Features/src/WebSocketAcceptContext.cs

```diff
@@ -14,5 +15,42 @@ public class WebSocketAcceptContext
         /// Gets or sets the subprotocol being negotiated.
         /// </summary>
         public virtual string? SubProtocol { get; set; }
+
+        /// <summary>
+        /// The interval to send pong frames. This is a heart-beat that keeps the connection alive.
+        /// </summary>
+        public virtual TimeSpan? KeepAliveInterval { get; set; }
+
+        /// <summary>
+        /// Enables support for the 'permessage-deflate' WebSocket extension.<para />
+        /// Be aware that enabling compression makes the application subject to CRIME/BREACH type attacks.
+        /// It is strongly advised to turn off compression when sending data containing secrets by
+        /// specifying <see cref="WebSocketMessageFlags.DisableCompression"/> when sending such messages.
+        /// </summary>
+        public bool DangerousEnableCompression { get; set; }
+
+        /// <summary>
+        /// Disables server context takeover when using compression.
+        /// </summary>
+        /// <remarks>
+        /// This property does nothing when <see cref="DangerousEnableCompression"/> is false,
+        /// or when the client does not use compression.
+        /// </remarks>
+        /// <value>
+        /// false
+        /// </value>
+        public bool DisableServerContextTakeover { get; set; }
+
+        /// <summary>
+        /// Sets the maximum base-2 logarithm of the LZ77 sliding window size that can be used for compression.
```

Comment:
Why would I set this property?

What is the valid range?

There should be argument validation on the setter.

Reply:
Also done